soroban-env-host/src/budget.rs

```diff
@@ -520,13 +536,26 @@ impl Default for BudgetImpl {
                     cpu.const_term = 1141;
                     cpu.lin_term = ScaledU64(1);
                 }
+                // We don't use a calibrated number for this because sending a
+                // large calibration-buffer to memcpy hits an optimized
+                // large-memcpy path in the stdlib, which has both a large
+                // overhead and a small per-byte cost. But large buffers aren't
+                // really how byte-copies usually get used in metered code. Most
+                // calls have to do with small copies of a few tens or hundreds
+                // of bytes. So instead we just "reason it out": we can probably
+                // copy 8 bytes per instruction on a 64-bit machine, and that
+                // therefore a 1-byte copy is considered 1/8th of an
+                // instruction. We also add in a nonzero constant overhead, to
+                // avoid having anything that can be zero cost and approximate
+                // whatever function call, arg-shuffling, spills, reloads or
+                // other flotsam accumulates around a typical memory copy.
```

Comment:
I've been thinking about consolidating some of those cost types, and put some thoughts in https://github.com/stellar/rs-soroban-env/issues/1020#issuecomment-1749621087

Some of these models have different linear coefficients, e.g. `HostMemCpy` vs. `HostMemCmp` which can be reasoned, and `ValSer` and `ValDeser` which is harder to reason about. 
Also they have different const coefficients, which will probably need to come from measure. 
So I was thinking a combination of measurement and reasoning to determine the proper overhead and scaling factor for the analytical model maybe the right thing to do? 
It can be a followup, don't want to block this PR. 

Another way that could work is in the calibration, copy/compare a small bytes array N times, instead of on one big array. This could avoid the AVX optimization? Personally I still prefer the analytical approach.

Reply:
Yeah, we can continue to iterate on this. I don't think this PR is the final word on the question; just what came to mind as I was doing review. I wanted to have a look at x86 (done so!) and do some eyeball-checks with tracy (done so!) and once I did those I wanted to shake some of the most obvious quirks out of what I found (memcpy numbers that don't make sense, plus the issue of overcounting VM setup obscuring undercounting the rest).